### PR TITLE
Fix for crashing E-RABSetupRequest with E-RAB ID > 12

### LINKED ISSUE #658: eNB crahes for VoLTE call when E-RAB ID > 12
## Issue Description ##
eNB crashes with the following logs, when placing 7th call from the same two UEs

```
Opening USRP channels=1, args: recv_frame_size=9232,send_frame_size=9232,type=b200,master_clock_rate=23.04e6
[INFO] [UHD RF] RF UHD Generic instance constructed
[INFO] [B200] Detected Device: B210
[INFO] [B200] Operating over USB 3.
[INFO] [B200] Initialize CODEC control...
[INFO] [B200] Initialize Radio control...
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Performing register loopback test... 
[INFO] [B200] Register loopback test passed
[INFO] [B200] Asking for clock rate 23.040000 MHz... 
[INFO] [B200] Actually got clock rate 23.040000 MHz.
Setting frequency: DL=2660.0 Mhz, UL=2540.0 MHz for cc_idx=0 nof_prb=25

==== eNodeB started ===
Type <t> to view trace
RACH:  tti=4211, cc=0, preamble=36, offset=0, temp_crnti=0x46
User 0x46 connected
User 0x46 connected
RACH:  tti=8431, cc=0, preamble=17, offset=0, temp_crnti=0x47
User 0x47 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
User 0x47 connected
User 0x46 connected
User 0x46 connected
User 0x47 connected
User 0x47 connected
User 0x46 connected
User 0x46 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
User 0x46 connected
User 0x47 connected
terminate called after throwing an instance of 'std::bad_alloc'
  what():  std::bad_alloc
srsRAN crashed... backtrace saved in './srsRAN.backtrace.crash'...
---  exiting  ---
srsRAN crashed... backtrace saved in './srsRAN.backtrace.crash'...
---  exiting  ---
terminate called recursively
srsRAN crashed... backtrace saved in './srsRAN.backtrace.crash'...
---  exiting  ---

```

## Setup Details ##
Running srsRAN enB (master) + open5gs EPC (tag v2.2.7) + kamailio IMS all in docker setup - https://github.com/herlesupreeth/docker_open5gs

SDR - USRP B210
UEs with sysmocom USIM/ISIM - Xiaomi phone and iPhone X

## Expected Behavior ##
Should not crash

## Actual Behaviour ##
eNB crashes the moment i attempt 7th call and works good for all the previous 6 calls

## Steps to reproduce the problem ##

- Run the open5gs EPC + IMS
- Run the srsRAN eNB
- Attach the two UEs to IMS
- UEs successfully register with IMS
- Make 6 calls between UEs
- Disconnect the call
- Place another (7th) call

## Additional Information ##
In order to allow E-RAB ID > 12 i.e. LCID > 10, I have done the following modification to srsRAN eNB. No other modification has been done

```
diff --git a/lib/include/srsran/common/common.h b/lib/include/srsran/common/common.h
index c0f5f5598..c341b24c3 100644
--- a/lib/include/srsran/common/common.h
+++ b/lib/include/srsran/common/common.h
@@ -40,8 +40,8 @@
 #define SRSRAN_UE_CATEGORY 4
 
 #define SRSRAN_N_SRB 3
-#define SRSRAN_N_DRB 11
-#define SRSRAN_N_RADIO_BEARERS 14
+#define SRSRAN_N_DRB 8
+#define SRSRAN_N_RADIO_BEARERS 11
 
 #define SRSRAN_N_MCH_LCIDS 32
```

I have attached the logs (INFO level), backtrace (taken with -DCMAKE_BUILD_TYPE=RelWithDebInfo), pcaps below. Please let me know if you need further details.
[log_bt_pcap.zip](https://github.com/srsran/srsRAN/files/6411021/log_bt_pcap.zip)

### PR DESCRIPTION
**Pull request by herlesupreeth**
Fixes #658. But also needs the fix in this PR - https://github.com/srsran/srsRAN/pull/657

Since number of Logical Channels are 0 to 10 (both inclusive) and only 3-10 are used for DRBs.
And, maximum DRBs an UE can have is 11 at any time. Therefore there is need to handle these DRBs
per UE with Logical Channels from 3 to 10. This commit address this issue by not relying on simple
calculation of LCID based on E-RAB ID (i.e. LCID = E-RAB - 2) or DRB ID. But rather keep track
of used LCID and assign unused ones when a new bearer needs to be created. If all LCIDs are used up
then eNB will respond to CN with a cause mentioning radio_res_not_available

### CODE REVIEW

**Comment by frankist** [comment]
Can you rebase this PR on https://github.com/srsran/srsRAN/tree/release_21_04_final . In the process, can you revert MAX_LC back to MAX_LC=11?

File: srsenb/src/stack/rrc/rrc_ue.cc [hunk 1]
```diff
@@ -1046,12 +1046,24 @@ bool rrc::ue::setup_erabs(const asn1::s1ap::erab_to_be_setup_list_ctxt_su_req_l&
 
 bool rrc::ue::release_erabs()
 {
+  for (const auto& erab_it : bearer_list.get_erabs()) {
+    bearer_list.rem_gtpu_bearer(erab_it.second.id);
```
Comment (frankist) [inline srsenb/src/stack/rrc/rrc_ue.cc:1050]: the `rem_gtpu_bearer` is now in `release_21_04_final` automatically called in `bearer_list.release_erabs()`, so this line can be removed.

File: lib/include/srsran/interfaces/sched_interface.h [hunk 2]
```diff
@@ -40,7 +40,7 @@ class sched_interface
 
   const static int MAX_SIB_PAYLOAD_LEN = 2048;
   const static int MAX_SIBS            = 16;
-  const static int MAX_LC              = 11;
+  const static int MAX_LC              = 14;
```
Comment (frankist) [inline lib/include/srsran/interfaces/sched_interface.h:43]: can you revert this change after you rebase

**Review by frankist — CHANGES_REQUESTED** [review]

File: srsenb/src/stack/rrc/rrc_ue.cc (hunk 1 above)
Comment (herlesupreeth) [reply srsenb/src/stack/rrc/rrc_ue.cc]: Done

File: lib/include/srsran/interfaces/sched_interface.h (hunk 2 above)
Comment (herlesupreeth) [reply lib/include/srsran/interfaces/sched_interface.h]: Done

**Comment by herlesupreeth** [comment]
@frankist Tried out this branch after rebasing with release_21_04_final branch and reverting MAX_LC to 11. eNB crashed when E-RAB setup request was issued with E-RAB ID 13

**Comment by herlesupreeth** [comment]
@frankist the latest commit fixes the crash. Test VoLTE calls for e-RAB Id > 12. Pcap attached
[test.zip](https://github.com/srsran/srsRAN/files/6467022/test.zip)
